### PR TITLE
completed issue map_nest_at_indices #2104

### PR DESCRIPTION
**Pull request by am02785**

### CODE REVIEW

**Comment by am02785** [comment]
This pull request resolves the issue map_nest_at_indices #2104,

Thank you

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi, 

Sorry for the late review, this PR flew under my radar,
Can you resolve the conflicts and also look into why the docstring is failing for this function.
Feel free to ask for any help ,I'd be happy to answer any question.

Thanks.

**Comment by am02785** [comment]
Hi sherry30, thank you for the message. I will try to resolve the conflicts and fix the docstring.

**Comment by am02785** [comment]
image

**Comment by am02785** [comment]
@sherry30 I have fixed the earlier error by replacing set_nest_at_indices with map_nest_at_indices, however I'm getting a new error and I'm not sure why: 
image

The following is part of the docstring: 
image

**Comment by am02785** [comment]
The example seems to work fine when I use docker: 
image

File: ivy/functional/ivy/nest.py [hunk 1]
```diff
@@ -312,7 +312,44 @@ def map_nest_at_indices(nest: Iterable, indices: Tuple, fn: Callable, /):
...
+    >>> ivy.map_nest_at_indices(nest, indices, fn)
+    >>> print(nest)
+    {'a': [8.0, 16.0, 23.0], 'b': [11.0, 44.0, 81.0], 'c': [9.0, 76.0, 37.0]}
+
+    With :code:`ivy.Array` inputs:
+
+    >>> nest = ivy.array([[-9., 8., -17.],[11., -3., 5.]])
+    >>> indices = ((0, 1),(1, 1),(1, 2))
+    >>> values = lambda x : x ** 2
+    >>> ivy.map_nest_at_indices(nest, indices, fn)
```
Comment (sherry30) [inline ivy/functional/ivy/nest.py:350]: replace `fn` with `values`

File: ivy/functional/ivy/nest.py [hunk 2]
```diff
@@ -312,7 +312,44 @@ def map_nest_at_indices(nest: Iterable, indices: Tuple, fn: Callable, /):
...
     fn
         The function to perform on the nest at the given index.
+    
+    Examples
+    --------
+    With :code:`List` inputs:
 
+    >>> nest = [['a', 'c', 'e', 'd', 'u', 'k'], ['m', 'n', 'f', 'p', 'q', 't']]
+    >>> indices = [[0, 4], [1, 5]]
+    >>> fn = lambda x : x + 'b'
```
Comment (sherry30) [inline ivy/functional/ivy/nest.py:322]: change `fn` to something else

File: ivy/functional/ivy/nest.py [hunk 3]
```diff
@@ -312,7 +312,44 @@ def map_nest_at_indices(nest: Iterable, indices: Tuple, fn: Callable, /):
...
+    >>> ivy.map_nest_at_indices(nest, indices, fn)
+    >>> print(nest)
+    [['a', 'c', 'e', 'd', 'ub', 'k'], ['m', 'n', 'f', 'p', 'q', 'tb']]
+
+    With :code:`Tuple` inputs:
+
+    >>> nest = ([-9, 8, -27],[9, -4, -5, 7])
+    >>> indices = ((0, 2),(1, 0),(1, 2))
+    >>> fn = abs
```
Comment (sherry30) [inline ivy/functional/ivy/nest.py:331]: change `fn`

File: ivy/functional/ivy/nest.py [hunk 4]
```diff
@@ -312,7 +312,44 @@ def map_nest_at_indices(nest: Iterable, indices: Tuple, fn: Callable, /):
...
+    >>> ivy.map_nest_at_indices(nest, indices, fn)
+    >>> print(nest)
+    ([-9, 8, 27], [9, -4, 5, 7])
+
+    With :code:`Dict` input:
+
+    >>> nest = {'a': [8., 16., 22.], 'b': [10., 44., 81.], 'c': [9., 75., 37.]}
+    >>> indices = (('a', 2), ('b', 0), ('c', 1))
+    >>> fn = lambda x : x + 1
```
Comment (sherry30) [inline ivy/functional/ivy/nest.py:340]: change `fn`

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

So i did some debugging and found out that you cannot use the argument `fn`, can you change its name to something else like, `fnn`, this argument is being used by our `docstring_test` function and it was being over written thats why it was giving the error.

Thanks.

**Comment by am02785** [comment]
@sherry30 Thank you for the response, I changed the name of the argument fn to function. I think that fixed the errors.

**Pull request merged**